app/pages/sections/order-items/associated-services/order-item/contextCreator.test.js

```diff
@@ -1,32 +1,69 @@
 import commonManifest from './commonManifest.json';
 import flatDeclarativeManifest from './flat/declarative/manifest.json';
 import flatOndemandManifest from './flat/ondemand/manifest.json';
-import { getContext, getErrorContext } from './contextCreator';
+import { backLinkHref, getContext, getErrorContext } from './contextCreator';
+import { baseUrl } from '../../../../../config';
 
 describe('associated-services order-item contextCreator', () => {
-  describe('getContext', () => {
-    it('should return the backLinkText', () => {
-      const context = getContext({
-        commonManifest,
-      });
-      expect(context.backLinkText).toEqual(commonManifest.backLinkText);
+  describe('backLinkHref', () => {
+    const orderId = 'order-id';
+    const associatedServicePrices = { prices: [{ priceId: 29 }, { priceId: 30 }] };
+    it('should return referer if referer ends in associated-service', () => {
+      const senderUrl = 'https://some.url.co.uk/order-id/items/associated-service';
+      const req = {
+        headers: {
+          referer: senderUrl,
+        },
+      };
+      const actual = backLinkHref({ req, orderId });
+
+      expect(actual).toEqual(senderUrl);
     });
 
-    it('should return the backLinkHref to associated-services when order item id is not neworderitem', () => {
-      const context = getContext({
-        commonManifest,
-        orderId: 'order-1',
-      });
-      expect(context.backLinkHref).toEqual('/order/organisation/order-1/associated-services');
+    it('should return referer if referer ends in price', () => {
+      const senderUrl = 'https://some.url.co.uk/order-id/select/price';
+      const req = {
+        headers: {
+          referer: senderUrl,
+        },
+      };
+      const actual = backLinkHref({ req, orderId });
+
+      expect(actual).toEqual(senderUrl);
     });
 
-    it('should return the backLinkHref to the select a price page when order item id is neworderitem', () => {
+    it('should return to associated-services url if referer ends with associated-services', () => {
+      const senderUrl = 'https://some.url.co.uk/order-id/select/associated-services';
+      const req = {
+        headers: {
+          referer: senderUrl,
+        },
+      };
+      const actual = backLinkHref({ req, orderId });
+
+      expect(actual).toEqual(senderUrl);
+    });
+
+    it('should return to associated-services price url if referer ends neworderitem and having more than 1 price', () => {
+      const associatedServicePriceUrl = `${baseUrl}/organisation/${orderId}/associated-services/select/associated-service/price`;
+      const senderUrl = 'https://some.url.co.uk/order-id/neworderitem';
+      const req = {
+        headers: {
+          referer: senderUrl,
+        },
+      };
+      const actual = backLinkHref({ req, associatedServicePrices, orderId });
+
+      expect(actual).toEqual(associatedServicePriceUrl);
+    });
+  });
```

Comment:
I think we can probably combine these 4 tests into a single test using `it.each`. There is an example of this in `app/helpers/common/priceFormatter.test.js`.

Reply:
Thanks Chris, I have refactored the test code related to backlinkHref.